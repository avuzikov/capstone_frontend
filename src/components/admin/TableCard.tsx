// src\components\admin\TableCard.tsx
import React from 'react';
import { Link } from 'react-router-dom';

interface ApplicantCardProps {
  link: string;
  name: string;
}

const TableCard = ({ link, name }: ApplicantCardProps) => {
  return (
    <Link to={link} className="w-full">
      <div className="card-bordered cursor-pointer flex items-center justify-between h-18">
        <div className="flex flex-col">
          <h1 className="text-medium">{name}</h1>
        </div>

        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          strokeWidth={3}
          stroke="currentColor"
          className="size-4"
        >
          <path strokeLinecap="round" strokeLinejoin="round" d="M9 5l7 7-7 7" />
        </svg>
      </div>
    </Link>
  );
};

export default TableCard;
