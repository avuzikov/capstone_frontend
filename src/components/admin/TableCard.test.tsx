// src\components\admin\TableCard.test.tsx

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import TableCard from './TableCard';

describe('TableCard Component', () => {
  const sampleProps = {
    link: '/admin/tables/users',
    name: 'Users',
  };

  test('renders the TableCard component with correct props', () => {
    render(
      <Router>
        <TableCard {...sampleProps} />
      </Router>
    );

    // Check for Link component with correct 'to' attribute
    const linkElement = screen.getByRole('link');
    expect(linkElement).toHaveAttribute('href', sampleProps.link);

    // Check for the name
    const nameElement = screen.getByText(sampleProps.name);
    expect(nameElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <Router>
        <TableCard {...sampleProps} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
