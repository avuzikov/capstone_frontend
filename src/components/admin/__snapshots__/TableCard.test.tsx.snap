// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCard Component matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="w-full"
    href="/admin/tables/users"
  >
    <div
      class="card-bordered cursor-pointer flex items-center justify-between h-18"
    >
      <div
        class="flex flex-col"
      >
        <h1
          class="text-medium"
        >
          Users
        </h1>
      </div>
      <svg
        class="size-4"
        fill="none"
        stroke="currentColor"
        stroke-width="3"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 5l7 7-7 7"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </a>
</DocumentFragment>
`;
